code/core/src/manager/components/sidebar/Tree.tsx

```diff
@@ -296,41 +326,99 @@ const Node = React.memo<NodeProps>(function Node({
   }
 
   if (item.type === 'component' || item.type === 'group') {
+    const { errors, warnings } = useStatusSummary(item);
+    const errorCount = Object.values(errors).length;
+    const warningCount = Object.values(warnings).length;
+
+    const itemStatus = groupStatus?.[item.id];
+    const color = itemStatus ? statusMapping[itemStatus][1] : null;
     const BranchNode = item.type === 'component' ? ComponentNode : GroupNode;
+
+    const createLinks: (onHide: () => void) => ComponentProps<typeof TooltipLinkList>['links'] = (
+      onHide
+    ) => {
+      const links = [];
+      if (errorCount) {
+        links.push({
+          id: 'errors',
+          icon: <CircleHollowIcon size={12} color={theme.color.negative} />,
+          title: `${errorCount} ${errorCount === 1 ? 'story' : 'stories'} with errors`,
+          onClick: () => {
+            const [firstStoryId, [firstError]] = Object.entries(errors)[0];
```

Comment:
is this kind of access safe? Should we add safe checks? same applies to line 360

Reply:
Yes it is, because we checked the error count first, so there's definitely a first entry. Combined with `useStatusSummary`'s return type, we are certain there will be an array of errors in the value here (with that first error we're certain to have).

Anyway I'll add some safeguards just in case any of these guarantees are lost in the future.